src/dispatch.js

```diff
@@ -0,0 +1,43 @@
+import {
+  sort,
+  comparator,
+  prop,
+  pipe,
+  head,
+  curryN,
+  defaultTo,
+  reduce,
+  reduced,
+  curry,
+  ifElse,
+} from 'ramda';
+
+import isNotNil from './isNotNil';
+import isNonEmptyArray from './isNonEmptyArray';
+import stubUndefined from './stubUndefined';
+
+const byArity = comparator((a, b) => a.length > b.length);
+
+const getMaxArity = pipe(
```

Comment:
I don't get why use `defaultTo` twice here ? 

IMHO, because we use `isNonEmptyArray` above, `getMaxArity` assume that take a non empty array and just be implemented like this :
```js
const getMaxArity = compose(length, head, sort(byArity))
```


---------------
In addition (just for informations), [R.curryN](https://github.com/ramda/ramda/blob/master/source/internal/_curryN.js) can be used with arity 0.

e.g.
```js
R.curryN(0, () => 42)() // => gives 42, no support for ramda placeholder
R.curryN(1, () => 42)() // => gives a Function, supports ramda placeholder
```

Reply:
That is true I overlooked this; pushing changes. Thanks

Do you guys still want to brainstorm on the name ?

Reply:
I cannot find any better name, `dispatch` is the best we have IMHO.